src/mono/mono/component/CMakeLists.txt

```
@@ -149,13 +149,13 @@ endif()
 foreach(component IN LISTS components)
   if(NOT DISABLE_COMPONENT_OBJECTS)
     add_library("${component}-objects" OBJECT "${${component}-sources}")
-    target_link_libraries("${component}-objects" component_base)
+    target_link_libraries("${component}-objects" PUBLIC component_base)
```

```suggestion
    target_link_libraries("${component}-objects" PRIVATE component_base)
```

I think `component_base` is an implementation detail of each component, so it should be private.